### PR TITLE
nodeinfo: add more nodeinfo

### LINKED ISSUE #65: feature: option to override the hostname
> btw, is there an option to override the hostname?

I don't think so, but with the config file it should be easilier possible to add such an option, since their is to need to add another argc. :smile:

_Originally posted by @herbetom in https://github.com/ffnord/mesh-announce/issues/60#issuecomment-669402551_

This could improve usability on cheap VPS hosts, where hostnames get changed on a regular basis by the hoster.

### PR DESCRIPTION
**Pull request by herbetom**
The following things can now be controlled domain specific:

Hostname, Hardware-Model, Contact Information, VPN, Latitude and Longitude

resolves #65

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -26,8 +28,14 @@ def __init__(self, name, parser, globals):
         '''
         from domain import Domain
 
+        self.contact = parser.get(name, 'Contact', fallback=None)
+        self.hardware_model = parser.get(name, 'Hardware-Model', fallback=None)
+        self.hostname = parser.get(name, 'Hostname', fallback=socket.gethostname())
         self.name = name
         self.interfaces = list(map(str.strip, parser.get(name, 'Interfaces', fallback='').split(',')))
+        self.is_gateway = (parser.get(name, 'VPN', fallback='True')).lower() in ['true', 'yes']
+        self.longitude = float(parser.get(name, 'Longitude', fallback=None))
```
Comment (herbetom) [inline config.py:37]: The fload conversion has to be done differently.

File: config.py [hunk 2]
```diff
@@ -26,8 +28,14 @@ def __init__(self, name, parser, globals):
...
+        self.latitude = float(parser.get(name, 'Latitude', fallback=None))
```
Comment (herbetom) [inline config.py:38]: The fload conversion has to be done differently.

**Comment by AiyionPrime** [comment]
Ah, I get it -.-'
`TypeError: float() argument must be a string or a number, not 'NoneType'`

**Comment by herbetom** [comment]
yeah, i would appreciate a saver and non error throwing suggestion for the conversion

**Comment by AiyionPrime** [comment]
What default values would you like `self.longitude` and `self.latitude` to have?

**Comment by AiyionPrime** [comment]
`None`?

**Comment by herbetom** [comment]
The Node should have no location if none is configured (like it is currently with mesh-announce or gluon Nodes). I don't know about the best approach to get to this result.

**Comment by AiyionPrime** [comment]
I'd go with either 
```
try:
            self.longitude = float(parser.get(name, 'Longitude', fallback=None))
        except TypeError:
            self.longitude = None
        try:
            self.latitude = float(parser.get(name, 'Latitude', fallback=None))
        except TypeError:
            self.latitude = None
```
Which follows your default in your example config;
https://github.com/Freifunk-Rhein-Neckar/mesh-announce/blob/d0739c1b1f614e98f8c03512cd7d96c031017898/respondd.conf.example#L33
https://github.com/Freifunk-Rhein-Neckar/mesh-announce/blob/d0739c1b1f614e98f8c03512cd7d96c031017898/respondd.conf.example#L36

Or this:
```
try:
            self.longitude = float(parser.get(name, 'Longitude', fallback=None))
        except TypeError:
            pass
        try:
            self.latitude = float(parser.get(name, 'Latitude', fallback=None))
        except TypeError:
            pass
```
Which I understood is the behavior described in your last comment?
Gotta go for today, thanks so far!

File: config.py (hunk 1 above)
Comment (herbetom) [reply config.py]: Conversion Errors are now caught: https://github.com/ffnord/mesh-announce/pull/63#issuecomment-778792117

File: config.py (hunk 2 above)
Comment (herbetom) [reply config.py]: Conversion Errors are now caught: https://github.com/ffnord/mesh-announce/pull/63#issuecomment-778792117

**Comment by herbetom** [comment]
I choose the first approach. The other didn't work (or had much noise about the missing Var).

@TobleMiner What do you think? Could this be merged?

**Comment by AiyionPrime** [comment]
Tested without the last force push; broken, if no coordinates were provided.
Tested with latest FP, works very well without them.

sn07 now runs with both #64 and #63 merged into master.

eg.
https://hannover.freifunk.net/karte/#/en/map/88e640ba7014

Without the hostname config parameter working it would identify as h2879555.stratoserver.net.

**Comment by AiyionPrime** [comment]
One possible enhancemnt I just cem across is configparsers `getfloat`method:
https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getfloat

**Comment by herbetom** [comment]
I did another Force Push to implement it with the `getfloat` method mentioned by @AiyionPrime 

I like that approach much more then catching exceptions and it seems to do exactly what is needed. :+1:

**Review by AiyionPrime — APPROVED** [review]
Tested yesterday before last forcepush.
Iirc there were only minor changes in the config comments.
The getbool method seems to be the cleanest solution.
Looks good to me.

**Pull request merged**